site/static/scripts/ui/domInteraction.js

```diff
@@ -0,0 +1,356 @@
+        ri = data.required_info;
+        populateAddDeviceForm(ri);
+    });
+}
+
+//Reads values from inputs and submits request for new device.
+function submitNewDevice(){
+    console.log("Adding new device");
+    var selectedPlugin = document.getElementById("pluginsSelect").value;
+    var selectedCollection = document.getElementById("serverCollectionsSelect").value;
+    var form = document.getElementById("addDeviceForm");
+    var inputs = form.getElementsByTagName("input");
+    var payload = {
+        "plugin_name": selectedPlugin,
+        "collection": selectedCollection,
+        "required_info": {}
+    };
+    //console.log(inputs);
+    for (var i = 0; i < inputs.length; i++){
+        if (inputs[i].type == "text"){
+            payload["required_info"][inputs[i].id] = inputs[i].value;
+        }
+    }
+    //console.log(payload);
+    postDevice(SELECTED_SERVER.address, payload);
+}
+
+function populateAddDeviceForm(requiredInfo){
+    var form = document.getElementById("addDeviceForm");
+    console.log(form);
+    removeChildren(form);
+    for (var infoType in requiredInfo){
+        var input = document.createElement("input");
+        var label = document.createElement("label");
+        input.type = "text";
+        input.id = infoType;
+        label.htmlFor = infoType;
+        label.appendChild(document.createTextNode(infoType));
+        form.appendChild(label);
+        form.appendChild(input);
+        form.appendChild(document.createElement("br"));
+    }
+    var submit = document.createElement("input");
+    submit.type = "submit";
+    submit.value = "Add";
+    form.appendChild(submit);
+}
+
+function populatePluginsSelect(plugins){
+    var pluginsSelect = document.getElementById("pluginsSelect");
+    removeChildren(pluginsSelect);
+    for (var i = 0; i < plugins.length; i++){
+        var opt = document.createElement("option");
+        var pName = plugins[i];
+        opt.value = pName;
+        opt.appendChild(document.createTextNode(pName));
+        pluginsSelect.appendChild(opt);
+    }
+}
+
+function populateServerCollectionsSelect(collections){
+    var collectionsSelect = document.getElementById("serverCollectionsSelect");
+    removeChildren(collectionsSelect);
+    for (var i = 0; i < collections.length; i++){
+        var opt = document.createElement("option");
+        var cName = collections[i];
+        opt.value = cName;
+        opt.appendChild(document.createTextNode(cName));
+        collectionsSelect.appendChild(opt);
+    }
+}
+
+//Generates the html to view the list of servers.
+function populateServers(servers, updateDevices){
+    var serversListElem = document.getElementById("serverNamesList");
+    removeChildren(serversListElem);
+
+    for (var name in servers){
+        // Create element for each device
+        var elem = document.createElement("li");
+        elem.className = "device_row";
+        elem.id = name;
+
+        // Creates button to delete server
```

Comment:
I think this button could be improved, it's rather big and doesn't really fit there.

Reply:
👍 The old GUI is entirely replaced this sprint. We are ensuring our GUI is easy to navigate and conforms to the topics from the lecture.